spacy_experimental/coref/coref_util.py

```diff
@@ -0,0 +1,276 @@
+from typing import List, Tuple, Dict, cast
+from thinc.types import Ints2d
+import srsly
+from spacy.language import Language
+from spacy.tokens import Doc
+import spacy.util as util
+
+# type alias to make writing this less tedious
+MentionClusters = List[List[Tuple[int, int]]]
+
+DEFAULT_CLUSTER_PREFIX = "coref_clusters"
+DEFAULT_CLUSTER_HEAD_PREFIX = "coref_head_clusters"
+
+
+@Language.factory(
+    "experimental_span_cleaner",
+    assigns=["doc.spans"],
+    default_config={"prefix": DEFAULT_CLUSTER_HEAD_PREFIX},
+)
+def make_span_cleaner(nlp: Language, name: str, *, prefix: str) -> "SpanCleaner":
+    """Create a span cleaner component.
+
+    Given a prefix, a span cleaner removes any spans on the Doc where the key
+    matches the prefix.
+    """
+
+    return SpanCleaner(prefix)
+
+
+class SpanCleaner:
+    def __init__(self, prefix: str):
+        self.cfg: Dict[str, Any] = {"prefix": prefix}
```

Comment:
`self.prefix` instead of `self.cfg`? (I think you can just leave `self.cfg` out without initializing it)

Reply:
You're right, that just worked. That's a relief, this seemed like a lot of boilerplate. I using the doc cleaner as a reference so I'm not sure what's different there...?

Reply:
Most of it is carry-over from v2-style component configs. I want to refactor it all for v4 so the settings aren't duplicated anywhere.

Reply:
But I'm trying to keep an eye on all the new stuff so that it is already okay.